model-server/src/main/kotlin/org/modelix/model/server/store/IgniteStoreClient.kt

```diff
@@ -14,29 +14,35 @@
  */
 package org.modelix.model.server.store
 
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
 import mu.KotlinLogging
 import org.apache.ignite.Ignite
 import org.apache.ignite.IgniteCache
 import org.apache.ignite.Ignition
 import org.modelix.kotlin.utils.ContextValue
-import org.modelix.model.IKeyListener
+import org.modelix.model.IGenericKeyListener
 import org.modelix.model.persistent.HashUtil
+import org.modelix.model.server.SqlUtils
 import java.io.File
 import java.io.FileReader
 import java.io.IOException
 import java.util.*
-import java.util.stream.Collectors
+import javax.sql.DataSource
 
 private val LOG = KotlinLogging.logger { }
 
-class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) : IStoreClient, AutoCloseable {
+class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) : IGenericStoreClient<ObjectInRepository>, AutoCloseable {
 
     companion object {
         private const val ENTRY_CHANGED_TOPIC = "entryChanged"
     }
 
     private lateinit var ignite: Ignite
-    private val cache: IgniteCache<String, String?>
+    private val legacyCache: IgniteCache<String, String?>
+    private val mutableObjectsCache: IgniteCache<ObjectInRepository, String?>
```

Comment:
`mutableObjectsCache` is currently a missleaing name.
It actuall might contain:

- global and repository-scoped mutable keys
- global **immutable** keys

See `igniteStoreClient.putAll`

Reply:
I reverted back to a single cache and decided that this is out of scope for this PR and MODELIX-866 should be fixed in a separate PR.